Pay chunkers fresh wallet

joshuef opened the pull request:
## Description

reviewpad:summary

Yoga07 commented on sn/src/dbs/kv_store/mod.rs:
```diff
@@ -97,96 +95,31 @@ impl<K: Key, V: Value + Send + Sync> KvStore<K, V> {
+        // Atomically write the value if it's new - this prevents multiple concurrent writes from
+        // consuming extra space (since sled is backed by a log). We throw away the value otherwise,
+        // since current use-cases do not require updates.
+        let _res = self
```
We could maybe log the result as before in the `match` below? Just to be wary that we aren't ignoring an error.

connec replied:
Yeah, I swithered on what logging to include. I think it's a good point to log it/match on it though, to make sure we don't lose the `?` in a refactor.

RolandSherwin committed:
fix(node): handling events should wait before connected to the network

joshuef committed:
feat!(protocol): remove chunk merkletree to simplify payment

starting to simplify payments to a per chunk basis

A first step in #638

joshuef committed:
feat!(protocol): get price and pay for each chunk individually

joshuef committed:
feat!(protocol): gets keys with GetStoreCost

joshuef committed:
feat!: pay each chunk holder direct

Sets us up to pay every node that will be storing the data.
We GetStoreCost and pay to their key, and put the chunk with the DBCs

This is not yet validated at the node side

joshuef committed:
feat(node): store data if the majority of CLOSE_GROUP will

joshuef committed:
test(node): reenable payment fail check

joshuef committed:
chore(node): clarify payment errors

joshuef committed:
chore(node): only store paid for data, ignore maj

joshuef committed:
chore(networking): return all GetStoreCost prices and use them

This gives maximal resilience as a place to start.
(This is required for data location verification tests eg)

joshuef committed:
chore(client): error out early for invalid transfers

joshuef committed:
feat: one transfer per data set, mapped dbcs to content addrs

joshuef committed:
test(node): data verification test refactors for readability

joshuef committed:
chore(node): data verification log tweaks

maqi committed:
fix: not check payment for relocated holder

maqi committed:
fix: correct replicated spend validation

RolandSherwin committed:
fix(network): trigger bootstrap until we have enough peers

joshuef committed:
ci: add missing package specification to dbc spend tests

joshuef committed:
ci: increase node mem limit with increased dbc processing

joshuef committed:
fix(protocol): avoid panics

joshuef committed:
chore: increase concurrent fetches for replication data

joshuef committed:
test: parallelise churn data final query

joshuef committed:
chore(networking): ensure we're always driving forward replication if pending

joshuef committed:
chore(client): reduce transferoutputs cloning

joshuef committed:
chore(client): pass around content payments map mut ref

joshuef committed:
feat: store dbcs to disk to reduce wallet burden
